File: appcompleta/frontend/src/types/stock.ts
export interface StockFilters {
  empresa_id?: string
  establecimiento_id?: string
  cultivo_id?: string
}

export interface StockSummary {
  stock_total_tn: number
  valor_stock: number
  unidades_alerta: number
  productos_criticos: number
  cobertura_promedio_dias: number
  costo_inventario_insumos: number
  necesidad_compra: number
  stock_inmovilizado: number
}

export interface StockItem {
  id: string
  cultivo: string
  establecimiento: string
  stock_disponible_tn: number
  vendido_tn: number
  comprometido_tn: number
  libre_tn: number
  libre_pct: number
  precio_referencia: number
  valor_estimado: number
  alerta: 'verde' | 'amarillo' | 'rojo'
}

export interface InsumoCoverageItem {
  insumo: string
  tipo: string
  unidad: string
  establecimiento: string
  stock_actual: number
  cobertura_dias: number
  cobertura_ha: number
  quiebre_en_dias: number | null
  compra_sugerida: number
  stock_inmovilizado: number
  costo_inventario: number
  consumo_plan_mes: number
  consumo_real_mes: number
  desvio_consumo_pct: number
  severidad: 'critical' | 'warning' | 'ok'
}

export interface ConsumoVsPlanItem {
  insumo: string
  consumo_plan_mes: number
  consumo_real_mes: number
  compra_sugerida: number
  desvio_pct: number
}

export interface StockAlert {
  id: string
  cultivo: string
  establecimiento: string
  mensaje: string
  severidad: 'warning' | 'critical' | 'info' | 'success'
}

export interface StockAction {
  titulo: string
  descripcion: string
  owner: string
  impacto: string
  severity: 'warning' | 'critical'
}

export interface StockData {
  summary: StockSummary
  items: StockItem[]
  insumos: InsumoCoverageItem[]
  consumo_vs_plan: ConsumoVsPlanItem[]
  alerts: StockAlert[]
  actions: StockAction[]
  ultima_actualizacion: string
}

File: appcompleta/backend/app/schemas/stock.py
"""
Schemas Pydantic para el modulo Stock.
"""
from pydantic import BaseModel
from typing import Optional, Any


class SummaryStock(BaseModel):
    stock_total_tn: float
    valor_stock: float
    unidades_alerta: int
    productos_criticos: int
    cobertura_promedio_dias: float
    costo_inventario_insumos: float
    necesidad_compra: float
    stock_inmovilizado: float


class StockItem(BaseModel):
    id: str
    cultivo: str
    establecimiento: str
    stock_disponible_tn: float
    vendido_tn: float
    comprometido_tn: float
    libre_tn: float
    libre_pct: float
    precio_referencia: float
    valor_estimado: float
    alerta: str


class InsumoItem(BaseModel):
    insumo: str
    tipo: str
    unidad: str
    establecimiento: str
    stock_actual: float
    cobertura_dias: float
    cobertura_ha: float
    quiebre_en_dias: Optional[int] = None
    compra_sugerida: float
    stock_inmovilizado: float
    costo_inventario: float
    consumo_plan_mes: float
    consumo_real_mes: float
    desvio_consumo_pct: float
    severidad: str


class StockResumenResponse(BaseModel):
    summary: SummaryStock
    items: list[StockItem]
    insumos: list[InsumoItem]
    consumo_vs_plan: list[Any]
    alerts: list[Any]
    actions: list[Any]
    ultima_actualizacion: str


class MovimientosResponse(BaseModel):
    total: int
    page: int
    page_size: int
    items: list[Any]
